add guide for using block specific frontend JavaScript

fabiankaegy opened the pull request:
This PR adds a new Guide to explaining how to best add frontend JavaScript to a block.

joesnellpdx reviewed (approved):
@fabiankaegy 

This is approved by me - gives basis for implementation - which is needed.

One thing we may want to consider as an enhancement - when a script is needed for a block AND a stand alone component (i.e. a custom modal block, but the modal is also used in a page template) - we'd want guidance for handling this use case as well.

Antonio-Laguna commented on guides/including-frontend-javascript-with-a-block.md:
```diff
@@ -0,0 +1,85 @@
+This automatically takes the JS file that is located at the relative file path and registers it using the `wp_register_script` function. The script gets the handle `namespace-example-view-script`. The handle is generated using the block namespace, followed by the block name, with the suffix `-view-script` added at the end.
+
+:::note
+WordPress expects and file that is provided via a relative file path to also have a `.asset.php` file next to it with the script dependencies and generated version number. Both `@wordpress/scripts` and `10up-toolkit` do this automatically for you using the `@wordpress/dependency-extraction-webpack-plugin`.
```
Is there a typo here?  `and file` should be `a file`?

fabiankaegy replied:
Fixed in https://github.com/10up/gutenberg-best-practices/commit/46dad8ab2293ff5ffaa0a3c5efc721aa88661fd8

Antonio-Laguna commented on guides/including-frontend-javascript-with-a-block.md:
```diff
@@ -0,0 +1,85 @@
+$asset_dependencies = ( include NAMESPACE_PATH . "dist/$asset_file_name.asset.php" );
+wp_register_script(
+	'my-custom-view-script-handle',
+	MAPS_BLOCK_APPLE_URL . "dist/$asset_file_name.js",
```
Should a more generic variable be used?

fabiankaegy replied:
Fixed in https://github.com/10up/gutenberg-best-practices/commit/07d9ccf462e9cc6b0c7563de42087866fb007f0a

Antonio-Laguna commented on guides/including-frontend-javascript-with-a-block.md:
```diff
@@ -0,0 +1,85 @@
+
+### Working with dynamic blocks
+
+Out of the box, this mechanism works great for static blocks that store their markup in the Database. However, most of the blocks we are building at 10up get built as dynamic blocks, using the PHP render_callback to create the markup on the server.
```
[nit] should we use `render_callback` (with code)?

fabiankaegy replied:
Fixed in https://github.com/10up/gutenberg-best-practices/commit/7180f0df3cb85eccf975b4d926675874deb022cd

Antonio-Laguna commented on guides/including-frontend-javascript-with-a-block.md:
```diff
@@ -0,0 +1,85 @@
+
+Out of the box, this mechanism works great for static blocks that store their markup in the Database. However, most of the blocks we are building at 10up get built as dynamic blocks, using the PHP render_callback to create the markup on the server.
+
+In these instances, WordPress will not automatically enqueue the script for us. So we manually need to add the `wp_enqueue_script` function to our `render_callback`. In the 10up theme scaffold this has been abstracted away into the `markup.php` file located in the blocks folder.
```
Should we add some link to the scaffold so it's easy to access?

fabiankaegy replied:
Fixed in https://github.com/10up/gutenberg-best-practices/commit/0566c016aa5dc44dded1b2d40abaf458839ccda5

Antonio-Laguna commented:
@fabiankaegy this looks awesome and it's a great explanation. I've left some really minor comments here. Feel free to ignore them! Can't wait to get this out!

fabiankaegy commented:
Thanks for the review @Antonio-Laguna I addressed your comments in followup commits since I had already merged and deployed the guide :) So all the items you called out should now be resolved and deployed aswell :)
